src/components/filters/MeasureValueFilter/DropdownBody.tsx

```diff
@@ -0,0 +1,126 @@
+import { IValue, MeasureValueFilterConditionOperator } from "./types";
+
+export interface IInputProps {
+    value?: IValue;
+    onChange: (value: IValue) => void;
+    onEnterKeyPress?: () => void;
+}
+
+export interface IDropdownBodyOwnProps {
+    operator?: MeasureValueFilterConditionOperator;
+    value?: IValue;
+    locale?: string;
+    onCancel: () => void;
+    onApply: (operator: MeasureValueFilterConditionOperator, value: IValue) => void;
+}
+
+type IDropdownBodyProps = IDropdownBodyOwnProps & InjectedIntlProps;
+
+interface IDropdownBodyState {
+    operator: MeasureValueFilterConditionOperator;
+    value: IValue;
+}
+
+class DropdownBodyWrapped extends React.PureComponent<IDropdownBodyProps, IDropdownBodyState> {
+    constructor(props: IDropdownBodyProps) {
+        super(props);
+
+        const { operator, value } = props;
+
+        this.state = {
+            operator,
+            value,
+        };
+    }
+
+    public render() {
+        const { onCancel, intl } = this.props;
+        const { operator } = this.state;
+
+        return (
+            <div className="gd-mvf-dropdown gd-dialog gd-dropdown overlay s-mvf-dropdown">
+                <div className="gd-mvf-dropdown-content">
+                    <div className="gd-mvf-dropdown-section">
+                        <OperatorDropdown onSelect={this.handleOperatorSelection} operator={operator} />
+                    </div>
+                    <div className="gd-mvf-dropdown-section">{this.renderInputSection()}</div>
+                </div>
+                <div className="gd-mvf-dropdown-footer">
+                    <Button
+                        className="gd-button-secondary gd-button-small s-mvf-dropdown-cancel"
+                        onClick={onCancel}
+                        value={intl.formatMessage({ id: "cancel" })}
+                    />
+                    <Button
+                        className="gd-button-action gd-button-small s-mvf-dropdown-apply"
+                        onClick={this.onApply}
+                        value={intl.formatMessage({ id: "apply" })}
+                        disabled={this.isApplyButtonDisabled()}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private renderInputSection = () => {
+        const { operator, value } = this.state;
+
+        if (Model.isComparisonTypeOperator(operator)) {
+            return (
+                <ComparisonInput
+                    value={value}
+                    onChange={this.handleValueChange}
+                    onEnterKeyPress={this.onApply}
+                />
+            );
+        } else if (Model.isRangeTypeOperator(operator)) {
+            return (
+                <RangeInput value={value} onChange={this.handleValueChange} onEnterKeyPress={this.onApply} />
+            );
+        }
+
+        return null;
+    };
+
+    private resetState = () => this.setState({ operator: undefined, value: undefined });
```

Comment:
In constructor, you copy the state values from props, here you set them to undefined. This seems inconsistent. Why do we need to support setting the value and operator from props in the first place?

Reply:
We need to be able to set component state from outside, for example, to update the existing measure value filter. I have removed reset function (it's not needed anyway now).

Reply:
Ok, makes sense :)